src/nbl/ext/DebugDraw/CDrawAABB.cpp

```diff
@@ -0,0 +1,429 @@
+// Copyright (C) 2018-2025 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#include "nbl/ext/DebugDraw/CDrawAABB.h"
+#include "nbl/builtin/hlsl/math/linalg/fast_affine.hlsl"
+
+#ifdef NBL_EMBED_BUILTIN_RESOURCES
+#include "nbl/ext/debug_draw/builtin/CArchive.h"
+#endif
+
+using namespace nbl;
+using namespace core;
+using namespace video;
+using namespace system;
+using namespace asset;
+using namespace hlsl;
+
+namespace nbl::ext::debug_draw
+{
+
+core::smart_refctd_ptr<DrawAABB> DrawAABB::create(SCreationParameters&& params)
+{
+	auto* const logger = params.utilities->getLogger();
+
+	if (!validateCreationParameters(params))
+	{
+		logger->log("Failed creation parameters validation!", ILogger::ELL_ERROR);
+		return nullptr;
+	}
+
+	smart_refctd_ptr<IGPUGraphicsPipeline> singlePipeline = nullptr;
+	if (params.drawMode & ADM_DRAW_SINGLE)
+	{
+		singlePipeline = createPipeline(params, params.singlePipelineLayout.get(), "single.vertex.hlsl", "aabb_instances.fragment.hlsl");
+		if (!singlePipeline)
+		{
+			logger->log("Failed to create pipeline!", ILogger::ELL_ERROR);
+			return nullptr;
+		}
+	}
+
+	smart_refctd_ptr<IGPUGraphicsPipeline> batchPipeline = nullptr;
+	if (params.drawMode & ADM_DRAW_BATCH)
+	{
+		batchPipeline = createPipeline(params, params.batchPipelineLayout.get(), "aabb_instances.vertex.hlsl", "aabb_instances.fragment.hlsl");
+		if (!batchPipeline)
+		{
+			logger->log("Failed to create pipeline!", ILogger::ELL_ERROR);
+			return nullptr;
+		}
+	}
+
+	if (!createStreamingBuffer(params))
+	{
+		logger->log("Failed to create streaming buffer!", ILogger::ELL_ERROR);
+		return nullptr;
+	}
+
+	auto indicesBuffer = createIndicesBuffer(params);
+	if (!indicesBuffer)
+	{
+		logger->log("Failed to create indices buffer!", ILogger::ELL_ERROR);
+		return nullptr;
+	}
+
+    return core::smart_refctd_ptr<DrawAABB>(new DrawAABB(std::move(params), singlePipeline, batchPipeline, indicesBuffer));
+}
+
+DrawAABB::DrawAABB(SCreationParameters&& params, core::smart_refctd_ptr<video::IGPUGraphicsPipeline> singlePipeline, smart_refctd_ptr<IGPUGraphicsPipeline> batchPipeline, smart_refctd_ptr<IGPUBuffer> indicesBuffer)
+    : m_cachedCreationParams(std::move(params)), m_singlePipeline(std::move(singlePipeline)), m_batchPipeline(std::move(batchPipeline)),
+    m_indicesBuffer(std::move(indicesBuffer))
+{
+}
+
+DrawAABB::~DrawAABB()
+{
+}
+
+// note we use archive entry explicitly for temporary compiler include search path & asset cwd to use keys directly
+constexpr std::string_view NBL_ARCHIVE_ENTRY = _ARCHIVE_ENTRY_KEY_;
```

Comment:
where does this variable come from and does everything still build without embedded resources cmake option?

Reply:
> where does this variable come

`_ARCHIVE_ENTRY_KEY_` is private define [set](<https://github.com/Devsh-Graphics-Programming/Nabla/pull/900/files/3b67580f67281905c2c6d694ed405148931841e0#diff-7094d64013982d99b5a328d8c33404abe36fe12a1cc754c7ce0b5ca0c2a8908fR29>) with CMake, scoped to the lib's sources only

> does everything still build without embedded resources cmake option?

builds and runs